src/SWP/Bundle/ContentBundle/Loader/ArticleLoader.php

```diff
@@ -79,8 +79,32 @@ public function load($type, $parameters = [], $withoutParameters = [], $response
 
             try {
                 $article = $this->articleProvider->getOneByCriteria($criteria);
-
-                return $this->getArticleMeta($article);
+                $articleBody = $article->getBody();
+                $metaArticle = $this->getArticleMeta($article);
+                //find local links in the article body
+                if (preg_match_all('/<a\s+href=["\']urn:newsml:([^"\']+)["\']/i', $articleBody, $links, PREG_PATTERN_ORDER)) {
```

Comment:
this logic should be in Meta Listeners - like this one: https://github.com/superdesk/web-publisher/blob/master/src/SWP/Bundle/CoreBundle/EventSubscriber/Meta/ArticleContentListsSubscriber.php

+ it must be cached (in memcached) - we can't do queries every time when article body with links is loaded.